packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -517,6 +559,7 @@ export class FeatureDevController {
                 }
             }
         }
+        await session.sendMetricDataTelemetry(MetricDataOperationName.END_CODE_GENERATION, MetricDataResult.SUCCESS)
```

Comment:
I'm not sure if these designs are final. But does it makes to just send a single metric instead of start/end? This would save some work on the server side